src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -352,16 +356,27 @@ export function makeUtxoEngineState(
           changeIndex: branch
         })
 
-        const { address: segwitAddress } = await internalGetFreshAddress({
+        const {
+          address: segwitAddress,
+          nativeBalance: segwitNativeBalance = '0'
+        } = await internalGetFreshAddress({
           ...commonArgs,
           format: getCurrencyFormatFromPurposeType(BIP43PurposeTypeEnum.Segwit),
           forceIndex,
           changeIndex: branch
         })
 
         return {
+          nativeBalance,
+          segwitNativeBalance,
```

Comment:
Sort these where the balance follows address for each type. Also this commit's message looks like it needs some minor cleanup.